structuralcodes/codes/mc2010/_concrete_shear.py

```diff
@@ -0,0 +1,1598 @@
+        alfa (float): Inclination of the stirrups
+
+    Returns:
+        The design shear resistance provided by shear reinforcement
+    """
+    if 45 < theta < 20:
+        warnings.warn("Too high or too low compression field angel")
+    return (
+        (asw / sw)
+        * z
+        * f_ywd
+        * ((1 / tan(theta * pi / 180)) + (1 / tan(alpha * pi / 180)))
+        * sin(alpha * pi / 180)
+    )
+
+
+def v_rd_max(
+    approx_lvl_s: int,
+    fck: float,
+    bw: float,
+    theta: float,
+    z: float,
+    E_s: float,
+    As: float,
+    Med: float,
+    Ved: float,
+    Ned: float,
+    delta_e: float,
+    alfa: float = 0,
+    gamma_c: float = 1.5,
+) -> float:
+    """The maximum allowed shear resistance, when there is shear reinforcment
+
+    fib Model Code 2010, eq. (7.3-24) and (7.3-26)
+
+    Args:
+        approx_lvl_s (int): Approximation level for steel
+        fck (float): Characteristic strength in MPa
+        z: (float): The length to the areasenter of cross-section in mm
+        bw: (float): Thickness of web in cross section
+        dg: (float): Maximum size of aggregate
+        E_s: (float): The E_s-modulus to the materialb in MPa
+        As: (float): The cross-section area in mm^2
+        Med: (float): The moment working on the material in Nmm
+        Ved: (float): The shear working on the material in N
+        Ned: (float): The normal force working on the material in N
+        delta_e (float): The exentricity of the load in mm
+        alfa (float): Inclination of the stirrups in degrees
+        gamma_c (float): Safety factor
+    Returns:
+        float: The maximum allowed shear resistance regardless of
+        approximation level"""
+    if approx_lvl_s == 1:
+        return v_rd_max_approx1(fck, bw, theta, z, alfa, gamma_c)
+
+    if approx_lvl_s == 2:
+        return v_rd_max_approx2(
+            fck, bw, theta, z, E_s, As, Med, Ved, Ned, delta_e, alfa, gamma_c
+        )
+
+    if approx_lvl_s == 3:
+        return v_rd_max_approx3(
+            fck, bw, z, E_s, As, Med, Ved, Ned, delta_e, alfa, gamma_c
+        )
+    raise ValueError("invalid approx level")
+
+
+def v_rd_max_approx1(
+    fck: float,
+    bw: float,
+    theta: float,
+    z: float,
+    alfa: float = 0,
+    gamma_c: float = 1.5,
+) -> float:
+    """The maximum allowed shear resistance, when there is shear reinforcment
+
+    fib Model Code 2010, eq. (7.3-37)
+
+    Args:
+        fck (float): Characteristic strength in MPa
+        z: (float): The length to the areasenter of cross-section in mm
+        bw: (float): Thickness of web in cross section
+        delta_e (float): The exentricity of the load in mm
+        alfa (float): Inclination of the stirrups in degrees
+        gamma_c (float): Safety factor
+
+    Returns:
+        float: The maximum allowed shear resistance regardless of
+        approximation level"""
+    nfc = min((30 / fck) ** (1 / 3), 1)
```

Comment:
Refactor out `nfc` to a separate function, since you use it several places. Consider naming it `eta_fc`.